CodeGeneration/Sources/SyntaxSupport/DeclNodes.swift

```diff
@@ -953,60 +999,83 @@ public let DECL_NODES: [Node] = [
     kind: .functionParameterList,
     base: .syntaxCollection,
     nameForDiagnostics: "parameter list",
+    documentation: """
+      A list of function parameters that are type annotated and may have an optional label.
+      The function parameters are represented by `FunctionParameterListSyntax`.
+
+      ### Example
+
+      ```swift
+      func foo(bar: Int, baz: Int) {
+
+      }
+      ```
+      """,
     elementChoices: [.functionParameter]
   ),
 
   Node(
     kind: .functionParameter,
     base: .syntax,
     nameForDiagnostics: "parameter",
+    documentation: "A function parameter",
     parserFunction: "parseFunctionParameter",
     traits: ["WithTrailingComma", "WithAttributes", "WithModifiers"],
     children: [
       Child(
         name: "attributes",
         kind: .collection(kind: .attributeList, collectionElementName: "Attribute", defaultsToEmpty: true),
-        nameForDiagnostics: "attributes"
+        nameForDiagnostics: "attributes",
+        documentation: "Attributes that are attached to the parameter."
       ),
       Child(
         name: "modifiers",
         kind: .collection(kind: .declModifierList, collectionElementName: "Modifier", defaultsToEmpty: true),
-        nameForDiagnostics: "modifiers"
+        nameForDiagnostics: "modifiers",
+        documentation: "Modifiers that are attached to the parameter."
       ),
       Child(
         name: "firstName",
-        kind: .token(choices: [.token(.identifier), .token(.wildcard)])
+        kind: .token(choices: [.token(.identifier), .token(.wildcard)]),
+        documentation: "The label of this parameter that will be used when the function is called."
       ),
       Child(
         name: "secondName",
         kind: .token(choices: [.token(.identifier), .token(.wildcard)], requiresLeadingSpace: true),
         nameForDiagnostics: "internal name",
+        documentation:
+          "If this is specified, it is the name by which the parameter can be referenced inside the function body.",
         isOptional: true
       ),
       Child(
         name: "colon",
-        kind: .token(choices: [.token(.colon)])
+        kind: .token(choices: [.token(.colon)]),
+        documentation: "If the parameter has a label, the colon separating the label from the type."
```

Comment:
`firstName` isn’t optional and neither is `colon`, so I think this should just be

```suggestion
        documentation: "The colon separating the label from the type."
```